### PR TITLE
Replaces the fedora in the DetDrobe with a noir fedora

### PR DESCRIPTION
**Pull request by AgentCitrus**
## About The Pull Request

The detective's fedora has a few special features (some damage resistance, the detective's flask, the ability to pick out candy corn) that the base fedora found in the DetDrobe doesn't have. This replaces the base fedora in the DetDrobe with a sprite variant of the detective's fedora (also adds a corgi hat version).

## Why It's Good For The Game

Now you can wear a full noir outfit without sacrificing the detective fedora's utilities.

## Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

image
image
image

</details>

## Changelog
:cl:
add: Replaces the fedora in the DetDrobe with a noir fedora, with the same abilities and armor as the base detective's hat
add: Hard-boiled Private Eye corgi variant
/:cl:

### CODE REVIEW

File: code/modules/vending/wardrobes.dm [hunk 1]
```diff
@@ -525,7 +525,7 @@
 					/obj/item/clothing/shoes/laceup = 2,
 					/obj/item/clothing/suit/det_suit/grey = 1,
 					/obj/item/clothing/suit/det_suit/noir = 1,
-					/obj/item/clothing/head/fedora = 2,
+					/obj/item/clothing/head/fedora/det_hat/noir = 2,
```
Comment (imsynthex) [inline code/modules/vending/wardrobes.dm:528]: Current change removes the default brown fedora from the vendor
```suggestion
					/obj/item/clothing/head/fedora/det_hat = 2,
					/obj/item/clothing/head/fedora/det_hat/noir = 2
```

Comment (AgentCitrus) [reply code/modules/vending/wardrobes.dm:528]: It does not, the original detective hat is further up the list of products

Comment (imsynthex) [reply code/modules/vending/wardrobes.dm:528]: Ah you're totally right, you should keep the 2 hats next to each other regardless for the sake of them showing up next to each other in the vendor

Comment (AgentCitrus) [reply code/modules/vending/wardrobes.dm:528]: They're grouped by outfit theme. Like the original fedora is with the brown trenchcoat, shoes, and suit, and the noir fedora is with the noir clothing and black laceup shoes.

Comment (imsynthex) [reply code/modules/vending/wardrobes.dm:528]: Alright, then code's solid

**Review by imsynthex — APPROVED** [review]

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**